src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/PostProcessingDocumentFilters/RemoveFailedGenerationOperationFilter.cs

```diff
@@ -23,14 +24,18 @@ public class RemoveFailedGenerationOperationFilter : IPostProcessingDocumentFilt
             OpenApiDocument openApiDocument,
             PostProcessingDocumentFilterSettings settings)
         {
-            if(openApiDocument == null || settings == null)
+            if (openApiDocument == null || settings == null)
             {
                 return;
             }
 
+            // Remove all operations with generation errors except for with DuplicateOperationException as duplicate for
+            // those operations are never added to the document.
             foreach (var operationDiagnostic in
                 settings.OperationGenerationDiagnostics.Where(
-                    operationDiagnostic => operationDiagnostic.Errors.Any()))
+                    operationDiagnostic => operationDiagnostic.Errors.Any()
+                    && operationDiagnostic.Errors.Any(i => i.ExceptionType
+                    != typeof(DuplicateOperationException).Name)))
```

Comment:
Is this necessary? Wouldn't the operation not be found for removal if it's never added to the document as the comment above says.

Reply:
no b/c the paths are duplicate it should document atleast one of them

---
In reply to: [192464495](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/130#discussion_r192464495) [](ancestors = 192464495)